src/nvim/api/buffer.c

```diff
@@ -457,7 +440,7 @@ void nvim_buf_set_lines(uint64_t channel_id, Buffer buffer, Integer start, Integ
   for (size_t i = to_replace; i < new_len; i++) {
     int64_t lnum = start + (int64_t)i - 1;
 
-    VALIDATE(lnum < MAXLNUM, "Index value is too high", {
+    VALIDATE_RANGE(lnum < MAXLNUM, "index", {
```

Comment:
Could be fancy and print `lnum < MAXLNUM` in the message, but decided to print `out of range` instead.